src/ECS/Logger.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+** EPITECH PROJECT, 2023
+** Raytracer
+** File description:
+** Logger.cpp
+*/
+
+#include "Logger.hpp"
+#include <ctime>
+#include <iostream>
+#include <map>
+#include <string>
+
+namespace Logger {
+    Logger::Logger(int logLevel) : _logLevel(logLevel)
+    {
+    }
+
+    void Logger::fatal(const std::string &message)
+    {
+        if (_logLevel < _LOG_FATAL_VALUE) {
+            return;
+        }
+        Logger::print(_LOG_FATAL_VALUE, "FATAL", message);
+    }
+
+    void Logger::error(const std::string &message)
+    {
+        if (_logLevel < _LOG_ERROR_VALUE) {
+            return;
+        }
+        Logger::print(_LOG_ERROR_VALUE, "ERROR", message);
+    }
+
+    void Logger::warn(const std::string &message)
+    {
+        if (_logLevel < _LOG_WARN_VALUE) {
+            return;
+        }
+        Logger::print(_LOG_WARN_VALUE, "WARN", message);
+    }
+
+    void Logger::info(const std::string &message)
+    {
+        if (_logLevel < _LOG_INFO_VALUE) {
+            return;
+        }
+        Logger::print(_LOG_INFO_VALUE, "INFO", message);
+    }
+
+    void Logger::debug(const std::string &message)
+    {
+#ifdef NDEBUG
+        return;
+#else
+        if (_logLevel < _LOG_DEBUG_VALUE) {
+            return;
+        }
+        Logger::print(_LOG_DEBUG_VALUE, "DEBUG", message);
+#endif
+    }
+
+    void Logger::trace(const std::string &message)
+    {
+#ifdef NDEBUG
+        return;
+#else
+        if (_logLevel < _LOG_TRACE_VALUE) {
+            return;
+        }
+        Logger::print(_LOG_TRACE_VALUE, "TRACE", message);
+#endif
+    }
+
+    void Logger::print(
+        int levelT,
+        const std::string &level,
+        const std::string &message)
+    {
+#ifdef __linux__
+        static std::map<int, std::string> colors = {
```

Comment:
Is it important to have this map in order?

Reply:
opti to use this